pallets/liquidity-pools/src/message.rs

```diff
@@ -530,9 +565,20 @@ pub enum Message<BatchContent = BatchMessages> {
 		/// The amount of tranche tokens which should be redeemed.
 		amount: u128,
 	},
+	/// A wrapped Message which was forwarded from the source domain via the
+	/// contract.
+	///
+	/// Directionality: Centrifuge <-> EVM Domain.
+	Forwarded {
+		source_domain: SerializableDomain,
+		forwarding_contract: H160,
+		message: ForwardContent,
+	},
```

Comment:
@hieronx This would be the (de-) serialization order. One downside of this generic approach for receiving and sending the same `Forwarded` message is that when Centrifuge Chain receives a forwarded message, `forwarding_contract` is not necessary. However, I believe it is feasible to just fill this `H160` with 0s on EVM side when forwarding.

Otherwise, we would have to implement two forwarded message types which increases complexity of (de-)serialization by orders of magnitude. 

A quick response would be highly appreciated! The rest of the PR are implementation details that you can skip.

Reply:
Yeah that seems like a good solution!